cacheOnReadFs.go

```diff
@@ -67,9 +67,8 @@ func (u *CacheOnReadFs) cacheStatus(name string) (state cacheState, fi os.FileIn
 	if err == syscall.ENOENT {
 		return cacheMiss, nil, nil
 	}
-	var ok bool
-	if err, ok = err.(*os.PathError); ok {
-		if err == os.ErrNotExist {
```

Comment:
Looks like you're in the right place, but this could use a couple tweaks.  I would remove the `if err == syscall.ENOENT {}` above and change your patch to:

```go
if err, ok := err.(*os.PathError); ok {
    if os.IsNotExist(err) {
```

`os.IsNotExist` will take care of syscall checks in a cross-platform way (`ENOENT` is unix-only).